### PR TITLE
SiteMiddleware and request.site are deprecated in wagtail 2.9

### PR DESCRIPTION
**Pull request by cwdavies**
Wagtail’s SiteMiddleware, which makes the current site object available as the property request.site, is now deprecated as it clashes with Django’s sites framework and makes unnecessary database queries on non-Wagtail views. References to request.site in your code should be removed; the recommended way of retrieving the current site is Site.find_for_request(request) in Python code, and the {% wagtail_site %} tag within Django templates. Once these are removed, 'wagtail.core.middleware.SiteMiddleware' can be removed from your project’s MIDDLEWARE setting.

https://docs.wagtail.io/en/stable/releases/2.9.html#sitemiddleware-and-request-site-deprecated

## Additions

- Site.find_for_request(request)

## Removals

- request.site

## How to test this PR

1. tox

## Checklist

_[Feel free to delete any checkboxes that are not applicable to this PR.]_

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing with the feature/section you're addressing, e.g., "Mega Menu: fix layout bug" or "Paying for College: Update content on Repay tool"
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Future todos are captured in comments and/or tickets
- [ ] Project documentation has been updated, potentialy one or more of:
  - [This repo’s docs](https://cfpb.github.io/cfgov-refresh/) (edit the files in the `/docs` folder) – for basic, close-to-the-code documentation on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing

_[When new or significantly modified front-end functionality is present, the following things should be tested. Feel free to delete this section if not applicable to this PR.]_

#### Browser testing

Visually tested in the following supported browsers:
- [ ] Firefox
- [ ] Chrome
- [ ] Safari
- [ ] Edge 18
- [ ] Internet Explorer 11 and 8 (via emulation in 11's dev tools)
- [ ] Safari on iOS
- [ ] Chrome on Android

[Further guidance on browser support](https://github.com/cfpb/development/blob/master/guides/browser-support.md)

#### Accessibility

- [ ] Keyboard friendly (navigable with tab, space, enter, arrow keys, etc.)
- [ ] Screen reader friendly
- [ ] Does not introduce new errors or warnings in [WAVE](https://wave.webaim.org/extension/)

#### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Does not introduce new lint warnings
- [ ] Flexible from small to large screens

### CODE REVIEW

File: cfgov/v1/util/util.py [hunk 1]
```diff
@@ -64,6 +66,11 @@ def get_secondary_nav_items(request, current_page):
             page.get_appropriate_siblings()
         )
 
+    if wagtail.VERSION > (2, 8):
```
Comment (chosak) [inline cfgov/v1/util/util.py:69]:
```suggestion
    if wagtail.VERSION >= (2, 9):
```

Wouldn't this line erroneously get hit on versions 2.8.1 or 2.8.2? Alternatively you could switch the logic around and use `if wagtail.VERSION < (2, 9):` as above.

Comment (willbarton) [reply cfgov/v1/util/util.py:69]: Also, to eliminate the coverage failure, you can add a `  # pragma: no cover` to this test (coverage is checked on `unittest-current`).

Comment (cwdavies) [reply cfgov/v1/util/util.py]: Updated to eliminate coverage failure and changed > (2, 8) to < (2, 9) with logic switch in `util.py`

**Comment by cwdavies** [comment]
@willbarton I removed `# pragma: no cover` around all version conditionals, and just put that on the `else` in `v1/util/util.py` that was causing the test coverage check fail.

**Pull request merged**